### DISCUSSION TITLE
run backend app before building the rest?

### DISCUSSION THREAD

**Discussion by HaruMiyaGi** [vercel/turborepo#662] [Help]
When running turborepo `yarn build` command the `getStaticPaths()` on my nextjs app errors because my backend is offline, so the function is unable to build correct paths.

The project is using turborepo, so when I run build command every app builds (including the backend)

I could just have my backend running in a separate tab when performing the build command which does work, however I would like it to be more robust.

Is there a way in turborepo to build my backend and then run it before starting to build all my other apps?

**Comment by clonecorp** [comment]
Same thing here. Even though I use the pipeline to have my API ready before the rest, it doesn't wait for the API server to be running before already building the rest. Did anyone find a solution to this?

In `turbo.json`

```
...
"pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".svelte-kit/**", ".vercel/**", "public/**"]
    },
    "app#build": {
      "dependsOn": ["api#preview"]
    },
    "www#build": {
      "dependsOn": ["api#preview"]
    },
...
```

**Comment by fabrice-reynaud-sosa** [comment]
Did anyone ever found a solution for this? Is the only way having the backend middleware handled in another project on an independent build and run cycle?
